src/core/util/sys_vars.cpp

```diff
@@ -157,6 +157,75 @@ const char *to_str(xlio_spec_t level)
 
 } // namespace xlio_spec
 
+namespace option_size {
+enum {
+    ONE_BYTE = 1U,
+    ONE_KB = 1024U,
+    ONE_MB = 1024U * 1024U,
+    ONE_GB = 1024U * 1024U * 1024U,
+};
+
+// Supported base: dec, hex, oct
+// Supported suffixes: one letter (K, M, G) and two letters (KB, MB, GB)
+//                     in lower or upper case
+//
+// All other formats or extra letters are considered as invalid input and
+// 0 is returned. Currently, there is no difference between 0 input and
+// a failure.
+size_t from_str(const char *str)
+{
+    char *endptr;
+    unsigned long val = strtoul(str, &endptr, 0);
+
+    struct size_suffix {
+        std::vector<const char *> vals;
+        unsigned long multiplier;
+    };
+    static const std::vector<size_suffix> suffixes = {{
+                                                          {"B", ""},
+                                                          ONE_BYTE,
+                                                      },
+                                                      {
+                                                          {"KB", "K"},
+                                                          ONE_KB,
+                                                      },
+                                                      {
+                                                          {"MB", "M"},
+                                                          ONE_MB,
+                                                      },
+                                                      {
+                                                          {"GB", "G"},
+                                                          ONE_GB,
+                                                      }};
+
+    for (auto &suffix : suffixes) {
+        for (auto sfx : suffix.vals) {
+            if (strcasecmp(sfx, endptr) == 0) {
```

Comment:
Don't we want to support lower case?

Reply:
With strcasecmp() we support any combination of lower and upper case.